examples/pxScene2d/src/pxResource.cpp

```diff
@@ -51,6 +51,12 @@ pxResource::~pxResource()
   {
     gUIThreadQueue->removeAllTasksForObject(this);
   }
+  if(((rtPromise*)mReady.getPtr())->status() == false)
```

Comment:
Is it possible for a pxResource to be destroyed if JavaScript code is still waiting on its promise?

Reply:
I would think not.